modules/extensions-mule-support/src/test/java/org/mule/runtime/module/extension/mule/internal/loader/parser/MuleSdkParameterModelParserSdkBuilder.java

```diff
@@ -0,0 +1,158 @@
+   *
+   * @param typeLoaderTypes The type mappings for the {@link TypeLoader}.
+   * @return This instance for chaining purposes.
+   */
+  public MuleSdkParameterModelParserSdkBuilder withTypeLoaderTypes(Map<String, MetadataType> typeLoaderTypes) {
+    this.typeLoaderTypes = typeLoaderTypes;
+    return this;
+  }
+
+  /**
+   * Allows for defining extension in context for the parameter parsing.
+   * <p>
+   * The given {@code extensionDeclarer} will be added to the set of {@link ExtensionDeclarer} which will later be loaded into
+   * {@link ExtensionModel} before building the parameter parser. No loading is done until the parameter parser is built.
+   *
+   * @param extensionDeclarer The {@link ExtensionDeclarer} to add to the parsing context.
+   * @return This instance for chaining purposes.
+   */
+  public MuleSdkParameterModelParserSdkBuilder withExtensionInContext(ExtensionDeclarer extensionDeclarer) {
+    extensionDeclarers.add(extensionDeclarer);
+    return this;
+  }
+
+  /**
+   * Adds a deprecation model to the {@link ComponentAst} to be used when creating the parameter parser.
+   * 
+   * @param since      String to return as the "since" parameter of the mock construct.
+   * @param message    String to return as the "message" parameter of the mock construct.
+   * @param toRemoveIn String to return as the "toRemoveIn" parameter of the mock construct.
+   * @return This instance for chaining purposes.
+   */
+  public MuleSdkParameterModelParserSdkBuilder withDeprecationModel(String since, String message, String toRemoveIn) {
+    deprecatedAst = mockDeprecatedAst(since, message, toRemoveIn);
+    return this;
+  }
+
+  /**
+   * Builds the {@link MuleSdkParameterModelParserSdk} using all the defined characteristics.
+   *
+   * @return The {@link MuleSdkParameterModelParserSdk}
+   */
+  public MuleSdkParameterModelParserSdk build() {
+    final TypeLoader typeLoader = mockTypeLoader(typeLoaderTypes != null ? typeLoaderTypes : emptyMap());
+    final ExtensionModelHelper extensionModelHelper = new ExtensionModelHelper(loadExtensionModels());
+
+    final ComponentAst componentAst = mock(ComponentAst.class);
+
+    final ComponentParameterAst parameterNameAst = stringParameterAst(name);
+    when(componentAst.getParameter(DEFAULT_GROUP_NAME, "name")).thenReturn(parameterNameAst);
+
+    final ComponentParameterAst typeParameterAst = stringParameterAst(type);
+    when(componentAst.getParameter(DEFAULT_GROUP_NAME, "type")).thenReturn(typeParameterAst);
+
+    if (deprecatedAst != null) {
+      setMockAstChild(componentAst, "deprecated", deprecatedAst);
+    }
+
+    return new MuleSdkParameterModelParserSdk(componentAst, typeLoader, extensionModelHelper);
+  }
+
+  private Set<ExtensionModel> loadExtensionModels() {
+    if (extensionDeclarers.isEmpty()) {
+      return emptySet();
+    }
+
+    return loadExtensionModels(createLoadingRequest(), new ExtensionModelFactory());
+  }
```

Comment:
For the `ExtensionModel` I actually found out that it was easier to load the real thing from the `ExtensionDeclarer` rather than mock everything. Mocking would have us messing too much with the internal hierarchy of models. Also we would have been missing possible tricks played by the enrichers.
Yes, the tests will be less "unitary", but I think that ship sailed anyway.